src/MarginTrading.SettingsService/Controllers/AssetPairsController.cs

```diff
@@ -264,5 +310,13 @@ private void ValidateId(string id, AssetPairContract contract)
                 throw new ArgumentException("Id must match with contract id");
             }
         }
+
+        private void ValidateUnique(AssetPairContract[] assetPairs)
+        {
+            if (assetPairs.Length == 0 || assetPairs.Select(x => x.Id).Count() != assetPairs.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(assetPairs), "Only unique asset pairs are allowed.");
```

Comment:
Error message is not clear. It is better to add something about ID )
And I think we should group by base, quote and legal entity to prevent error if batch contains duplicates.